core/chains/evm/client/pool.go

```diff
@@ -122,16 +124,15 @@ func (p *Pool) Dial(ctx context.Context) error {
 				rawNode.nLiveNodes = p.nLiveNodes
 			}
 			// node will handle its own redialing and automatic recovery
-			if err := n.Start(ctx); err != nil {
+			if err := ms.Start(ctx, n); err != nil {
```

Comment:
What does this get us?

Reply:
> If any service fails to start, the previously started services will be Closed

If any node after the first failed to start, previously started nodes were left orphaned.